cmd/iam-store.go

```diff
@@ -1583,6 +1583,10 @@ func (store *IAMStoreSys) UserNotificationHandler(ctx context.Context, accessKey
 		m = cache.iamUsersMap
 	}
 	err := store.loadUser(ctx, accessKey, userType, m)
+	defer func() {
+		cache.updatedAt = time.Now()
```

Comment:
It makes sense for me to only update cache.updateAt when this call returns a nil error, /cc @poornas

Reply:
yeah that makes sense @donatello ^^

Reply:
That's not completely correct - we do in some cases update the cache and still return non-nil error. I am slightly updating this logic.